controllers/spaceprovisionerconfig/mapper_test.go

```diff
@@ -0,0 +1,63 @@
+package spaceprovisionerconfig
+
+import (
+	"context"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestToolchainClusterToSpaceProvisionerConfigMapper(t *testing.T) {
+	// given
+	cl := test.NewFakeClient(t, &toolchainv1alpha1.SpaceProvisionerConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "spc0",
+			Namespace: test.HostOperatorNs,
+		},
+		Spec: toolchainv1alpha1.SpaceProvisionerConfigSpec{
+			ToolchainCluster: "cluster1",
+		},
+	}, &toolchainv1alpha1.SpaceProvisionerConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "spc1",
+			Namespace: test.HostOperatorNs,
+		},
+		Spec: toolchainv1alpha1.SpaceProvisionerConfigSpec{
+			ToolchainCluster: "cluster2",
+		},
+	}, &toolchainv1alpha1.SpaceProvisionerConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "spc2",
+			Namespace: test.HostOperatorNs,
+		},
+		Spec: toolchainv1alpha1.SpaceProvisionerConfigSpec{
+			ToolchainCluster: "cluster1",
+		},
+	})
+
+	// when
+	reqs, err := findReferencingProvisionerConfigs(context.TODO(), cl, runtimeclient.ObjectKey{Name: "cluster1", Namespace: test.HostOperatorNs})
```

Comment:
Let's test 4 cases:
1. when there is one match
2. when there are multiple matches
3. when there is no match
4. when the list fails

we can theoretically join 1. and 2. together, but if it's not a big deal, then it's nice to have it separate.

Reply:
done